app/src/main/java/com/concordia/mcga/adapters/POISearchAdapter.java

```diff
@@ -0,0 +1,283 @@
+import android.view.ViewGroup;
+import android.widget.BaseExpandableListAdapter;
+import android.widget.TextView;
+
+import com.concordia.mcga.activities.R;
+import com.concordia.mcga.models.Building;
+import com.concordia.mcga.models.Campus;
+
+import java.util.ArrayList;
+
+import de.hdodenhof.circleimageview.CircleImageView;
+
+/**
+ * This adapter searches through both campuses to find POIs (for now, buildings) that match a query.
+ */
+public class POISearchAdapter extends BaseExpandableListAdapter {
+    private Context context;
+
+    public static final int SGW_INDEX = 0;
+    public static final int LOYOLA_INDEX = 1;
+
+    private ArrayList<Building> sgwFilteredList;
+    private ArrayList<Building> loyolaFilteredList;
+
+    /**
+     * The constructor initializes the empty lists that store the currently queried POIs
+     * @param context View context to search
+     */
+    public POISearchAdapter(Context context) {
+        this.context = context;
+
+        this.sgwFilteredList = new ArrayList<>();
+        this.loyolaFilteredList = new ArrayList<>();
+    }
+
+    /**
+     * Evaluates the number of campuses active based on the current queried POIs
+     * @return The number of active campuses
+     */
+    @Override
+    public int getGroupCount() {
+        boolean sgwClear = getGroupIsEmpty(SGW_INDEX);
+        boolean loyClear = getGroupIsEmpty(LOYOLA_INDEX);
+
+        if (sgwClear && loyClear) {
+            return 0;
+        } else if (sgwClear || loyClear) {
+            return 1;
+        } else {
+            return 2;
+        }
+    }
+
+    /**
+     * This overrides the BaseExpandableList function to get the number of children (POI) entries
+     * @param i Campus index
+     * @return Number of POIs specific to the passed campus
+     */
+    @Override
+    public int getChildrenCount(int i) {
+        Campus camp = (Campus)getGroup(i);
+        if (camp == Campus.SGW) {
+            return sgwFilteredList.size();
+        } else if (camp == Campus.LOY) {
+            return loyolaFilteredList.size();
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * This overrides the BaseExpandableList function to get the campus specified by an index.
+     * This function is written specifically to address the quirks in the way BaseExpandableList
+     * handles groups and children - it is used internally to get the "group" (campus) based on the
+     * number of visible groups, so we have to handle different cases depending on whether the
+     * POIs filtered by the query come from a single campus or both of them.
+     * @param i Active campus index
+     * @return The campus specified by the index
+     */
+    @Override
+    public Object getGroup(int i) {
+        boolean sgwClear = getGroupIsEmpty(SGW_INDEX);
+        boolean loyClear = getGroupIsEmpty(LOYOLA_INDEX);
+
+        if (sgwClear && loyClear) {
+            return null;
+        } else if (sgwClear) {
+            if (i == 0) {
```

Comment:
I'm almost sure this should be a **1** not a 0, or even better, use the class constants `SGW_INDEX , LOY_INDEX` you had defined above.